sdk/communication/Azure.Communication.Calling.Server/src/Models/CallState.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ComponentModel;
+using System.IO;
+
+namespace Azure.Communication.Calling.Server
+{
+    /// <summary>
+    /// The states of a call.
+    /// </summary>
+    public readonly struct CallState : IEquatable<CallState>
```

Comment:
We should be using extensible enum instead of manually converting strings to enums

Reply:
I found this as a example of extensible enum, https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/objectanchors/Azure.MixedReality.ObjectAnchors.Conversion/src/AssetFileType.cs
And doing the same strcture. 
Can you please elaborate what required to make CallState as a extensible enum?

Reply:
Addressed in new Iteration